src/aiidalab_qe/workflows/__init__.py

```diff
@@ -141,10 +145,35 @@ def get_builder_from_protocol(
                     hubbard_type="U",
                     use_kinds=True,
                 )
-            hubbard_structure.store()
-            builder.structure = hubbard_structure
+
+            # Initialize flag to determine when to store hubbard_structure
+            should_store_and_assign = False
+
+            if not isinstance(structure, HubbardStructureData):
+                # Store the HubbardStructureData is structure is not a HubbardStructureData
+                should_store_and_assign = True
+            else:
+                # Check if hubbard parameters are different from the structure
+                if hubbard_structure.hubbard != structure.hubbard:
+                    should_store_and_assign = True
+                else:
+                    # If criteria is not met, assign the structure to the builder
+                    builder.structure = structure
+
+            # Common logic to store and assign hubbard_structure if conditions are met
+            if should_store_and_assign:
+                hubbard_structure.store()
+                builder.structure = hubbard_structure
```

Comment:
this logic can be simplified.
```suggestion
            # If this is a new HubbardStructureData or the hubbard is different
            if not isinstance(structure, HubbardStructureData) or hubbard_structure.hubbard != structure.hubbard:
                hubbard_structure.store()
                builder.structure = hubbard_structure
            else:
                builder.structure = structure
```